src/audio/generic_processor/generic_processor.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+
+/*
+ * A generic processor component.
+ */
+
+/**
+ * \file audio/generic_processor.c
+ * \brief Processing compoent aimed to work with external codecs
+ * \author Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ */
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/generic_processor/generic_processor.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/drivers/ipc.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/clk.h>
+#include <sof/lib/memory.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/ut.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <user/trace.h>
+
+static const struct comp_driver comp_generic_processor;
+
+//TODO: provide proper uuid below
+/* d8218443-5ff3-4a4c-b388-6cfe07b956aa */
```

Comment:
Is this file meant to be a template to be used by other processing components ? If so, we should state more details of how to use this file in the top level comments.

Reply:
@lgirdwood I will add some more comments there sure, nevertheless we need a dedicated page in sof-docs for this component